MsixCore/msixmgr/msixmgr.cpp

```diff
@@ -565,47 +941,91 @@ int main(int argc, char * argv[])
+
                 return ERROR_NOT_SUPPORTED;
             }
             return S_OK;
         }
         case OperationType::UnmountImage:
         {
+            // Telemetry : Unmount Workflow Log
+            msixmgr::TraceLogUnmountWorkflow(workflowId.c_str(), cli.GetFileTypeAsString().c_str());
+
             WVDFileType fileType = cli.GetFileType();
             if (fileType == WVDFileType::CIM)
             {
                 if (cli.GetVolumeId().empty() && cli.GetMountImagePath().empty())
                 {
+                    errorCode = msixmgr::GetErrorCodeFromHRESULT(E_INVALIDARG);
+                    errorDesc = L"Neither CIM file path Argument given nor VolumeId of the image to be unmounted given. HRESULT " + errorCode + L".";
+
                     std::wcout << std::endl;
                     std::wcout << "To unmount an CIM image, please provide either the CIM file path or the volume the image was mounted to." << std::endl;
                     std::wcout << "The CIM file path can be specified using the -imagepath option." << std::endl;
                     std::wcout << "The volume can be specified using the -volumeId option." << std::endl;
                     std::wcout << std::endl;
+
+                    // Telemetry : Workflow Log
+                    QueryPerformanceCounter(&msixMgrLoad_EndCounter);
+                    workflowElapsedTime = msixmgr::CalcWorkflowElapsedTime(msixMgrLoad_StartCounter, msixMgrLoad_EndCounter, msixMgrLoad_Frequency);
+                    msixmgr::TraceLogWorkflow(workflowId.c_str(), cli.GetOperationTypeAsString().c_str(), false, workflowElapsedTime, errorCode.c_str(), errorDesc.c_str());
+
                     return E_INVALIDARG;
                 }
 
                 HRESULT hrUnmountCIM = MsixCoreLib::UnmountCIM(cli.GetMountImagePath(), cli.GetVolumeId());
 
                 if (FAILED(hrUnmountCIM))
                 {
+                    errorCode = msixmgr::GetErrorCodeFromHRESULT(hrUnmountCIM);
+                    errorDesc = L"Failed Unmounting the CIM file. HRESULT " + errorCode + L".";
+
                     std::wcout << std::endl;
                     std::wcout << "Unmounting the CIM " << " failed with HRESULT 0x" << std::hex << hrUnmountCIM << std::endl;
 
                     // ERROR_NOT_FOUND may be returned if only the mount image path but not the volume id was provided
                     // and msixmgr was unable to find the volume id associated with a given image path.
                     if (hrUnmountCIM == HRESULT_FROM_WIN32(ERROR_NOT_FOUND) && cli.GetVolumeId().empty())
                     {
+                        errorDesc += L"Failure while finding the VolumeId associated with the given image path. Can be retried directly using the VolumeId option";
+
                         std::wcout << "The error ERROR_NOT_FOUND may indicate a failure to find the volume id associated with a given image path."<< std::endl;

```

Comment:
Same as above

Reply:
As discussed, all the Telemetry variables will have the same Error Message as the actual Console statements. The change for outputting the Error Message on Console from a variable will be done as part of Error Message Improvement Task that will be taken later.